app/models/group.rb

```diff
@@ -31,8 +31,41 @@ def roles
     @roles ||= Group.find(uuid).roles
   end
 
-  def has_role?(role_name)
-    roles.any? { |role| role.name == role_name }
+  def can_approve?
+    context = Insights::API::Common::Request.current.to_h.transform_keys(&:to_s)
+    ContextService.new(context).as_org_admin do
+      (scopes & ['admin', 'group']).any?
+    end
+  end
+
+  private
+
+  def scopes
```

Comment:
rename `action_create_scopes` because you are collecting scopes for specific permissions only